### PR TITLE
Best way to add_msvc_runtime_flag

### PR DESCRIPTION
**Pull request by lygstate**
Best way to implement add_msvc_runtime_flag for choose msvc runtime properly.
Refer to https://stackoverflow.com/questions/24460486/cmake-build-type-not-being-used-in-cmakelists-txt
So that cmake_args doesn't need to append build type. we choose the build type at the build phase.

### CODE REVIEW

**Comment by zrphercule** [comment]
@onnxbot retest this please

**Comment by lygstate** [comment]
Do not know why it's failed. weird.

File: setup.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def run(self):
             os.makedirs(CMAKE_BUILD_DIR)
 
         with cd(CMAKE_BUILD_DIR):
+            build_type = 'Release'
```
Comment (bddppq) [inline setup.py]: I think cmake's default build type is not necessary "Release"?

File: cmake/Utils.cmake [hunk 2]
```diff
@@ -2,17 +2,9 @@
 # Add MSVC RunTime Flag
 function(add_msvc_runtime_flag lib)
   if(${ONNX_USE_MSVC_STATIC_RUNTIME})
-    if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
-      target_compile_options(${lib} PRIVATE /MTd)
-    else()
-      target_compile_options(${lib} PRIVATE /MT)
-    endif()
+    target_compile_options(${lib} PRIVATE $<$<NOT:$<CONFIG:Debug>>:/MT> $<$<CONFIG:Debug>:/MTd>)
```
Comment (bddppq) [inline cmake/Utils.cmake]: What's the benefit of changing to use generator expression for this? IMO this new code looks more cryptic than the old version.

File: setup.py (hunk 1 above)
Comment (bddppq) [reply setup.py]: @raymondxyang `build_type` will be unconditionally appended to `cmake_args`

File: cmake/Utils.cmake (hunk 2 above)
Comment (lygstate) [reply cmake/Utils.cmake]: https://stackoverflow.com/questions/34490294/what-does-configdebugrelease-mean-in-cmake

File: setup.py (hunk 1 above)
Comment (lygstate) [reply setup.py]: Refer to https://stackoverflow.com/questions/24460486/cmake-build-type-not-being-used-in-cmakelists-txt

Comment (lygstate) [reply setup.py]: After https://stackoverflow.com/questions/24460486/cmake-build-type-not-being-used-in-cmakelists-txt, there is no need apped build_type to cmake_args.

**Comment by lygstate** [comment]
@zrphercule Please take a look at this and merge this,

**Pull request closed without merging**